sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregators/BaseAggregator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics.aggregators;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/** BaseAggregator represents the base class for all the available aggregations. */
+@ThreadSafe
+public interface BaseAggregator {
+
+  /**
+   * Merge aggregated values between the current instance and the given {@code aggregator}.
+   *
+   * @param aggregator value to merge with.
+   */
+  void merge(BaseAggregator aggregator);
```

Comment:
What would you think about adding a type parameter like 
`public interface BaseAggregator<T extends BaseAggregator<?>>` and then having this method take `T` as the parameter? Then, you can implement exactly what you can merge with in the concrete implementations, rather than having to check for the self-type on every merge call.